server/routes/volunteers.ts

```diff
@@ -144,6 +144,441 @@ router.route("/api/volunteers/").get(
   })
 );
 
+// Utility function to check if the volunteer type has changed
+function hasVolunteerTypeChanged(
+  originalType: string[],
+  newType: string[]
+): boolean {
+  if (originalType.length !== newType.length) return true;
+
+  return originalType.length === 1 && newType[0] !== originalType[0];
+}
+
+/**
+ * @description Update a volunteers info
+ * @route  PATCH /api/volunteers/update/:volunteerId
+ * @access
+ */
+router.route("/api/volunteers/update/:volunteerId").patch(
+  adminProtect,
+  asyncHandler(async (req: Request, res: Response) => {
+    const { volunteerId } = req.params;
+    logger.info(`PATCH /api/volunteers/update/${volunteerId}`);
+    logger.info("Request body: ", req.body);
+
+    const {
+      firstName,
+      lastName,
+      email,
+      phoneNumber,
+      participantType,
+    }: {
+      firstName: string;
+      lastName: string;
+      email: string;
+      phoneNumber: string;
+      participantType: string[];
+    } = req.body;
+
+    const stringFields = [firstName, lastName, email, phoneNumber];
+
+    const isValidRequest = stringFields.every((field) => {
+      return field.trim().length > 0;
+    });
+
+    if (!isValidRequest) {
+      logger.error(`Invalid request body: ${JSON.stringify(req.body)}`);
+
+      res.status(BAD_REQUEST).json({
+        message: `Please provide a valid 'firstName', 'lastName', 'email', and 'phoneNumber' on the body.`,
+      });
+    }
+
+    // also verify that the participantType is valid
+    const validParticipantTypes = ["Driver", "Packer"];
```

Comment:
Since these are basically types, I personally would suggest doing something like this for better type safety and ease of understanding (since Typescript will tell us that these are "Driver" or "Distributor" instead of just "string" types. It would also probably be better to have these types in the frontend where we make this request. 
```ts
export const AIRTABLE_PARTICIPANT_TYPES = ["Driver", "Distributor"] as const;
export type AirtableParticipantType =
  (typeof AIRTABLE_PARTICIPANT_TYPES)[number];
export const PARTICIPANT_TYPES = ["Driver", "Packer"] as const;
export type ParticipantType = (typeof PARTICIPANT_TYPES)[number];
// ^ Probably could be put in the types.ts file

    const isParticipantTypeValid =
      participantType.length > 0 &&
      participantType.every((type) => PARTICIPANT_TYPES.includes(type));

    if (!isParticipantTypeValid) {
      res.status(BAD_REQUEST).json({
        message: `Please provide a valid 'participantType' on the body, valid types are ${PARTICIPANT_TYPES}. 
        Was provided ${participantType}.`,
      });
    }

    // replace all Packer with Distributor the array for airtable
    const airtableParticipantType = participantType.map(
      (type: ParticipantType): AirtableParticipantType => {
        return type === "Packer" ? "Distributor" : type;
      }
    );
```

Reply:
I would also probably check that the length of the participant type array in the request is <= 2, since we are kinda making that assumption elsewhere